src/workerd/api/tracing-module.c++

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) 2017-2025 Cloudflare, Inc.
+// Licensed under the Apache 2.0 license found in the LICENSE file or at:
+//     https://opensource.org/licenses/Apache-2.0
+
+#include "tracing-module.h"
+
+namespace workerd::api {
+
+JsSpan::JsSpan(kj::Maybe<IoOwn<SpanBuilder>> span): span(kj::mv(span)) {}
+
+JsSpan::~JsSpan() noexcept(false) {
+  end();
```

Comment:
There's a chance that the JsSpan could outlive the IoContext, in which case the `s->end()` in the `end()` call will throw an error. Is that what you want here?